public/app/features/explore/Explore.tsx

```diff
@@ -497,6 +500,14 @@ export class Explore extends React.PureComponent<Props, ExploreState> {
         queryResponse.customFrames,
       ].every((e) => e.length === 0);
 
+    let correlationsBox = undefined;
+    const isCorrelationsEditorMode = correlationEditorDetails?.editorMode;
+    const showCorrelationHelper =
+      isCorrelationsEditorMode || (!isCorrelationsEditorMode && correlationEditorDetails?.dirty) || false;
```

Comment:
this should be equivalent:
```suggestion
    const showCorrelationHelper = Boolean(isCorrelationsEditorMode || correlationEditorDetails?.dirty);
```